torch/csrc/jit/script/compiler.cpp

```diff
@@ -1360,20 +1366,55 @@ struct to_ir {
     return num_starred;
   }
 
-  void emitAssignment(const Assign& stmt) {
-    bool starred_unpack = calcNumStarredUnpack(stmt.lhs(), stmt.range());
-    if (stmt.reduction() != '=') {
-      if (stmt.lhs().size() != 1) {
-        throw ErrorReport(stmt)
-            << "reductions are only allowed when there is a single variable "
-            << "on the left-hand side.";
+  // Emit nodes for augmented assignments like `+=`
+  void emitAugAssignment(const AugAssign& stmt) {
+    auto lhs = Var(stmt.lhs());
```

Comment:
I don't think anything forces this to be a var. So this could be a bad error message if someone writes `a[b] += c`. However, I know you have a followup PR that will handle this case, so It is probably ok to fix it there, especially since the earlier version of this code had the same bug.

Reply:
Yeah the [next PR](https://github.com/pytorch/pytorch/pull/13486) changes this but I was too lazy to absorb the changes into this one :/